### DISCUSSION TITLE
微调使达到官方标准

### DISCUSSION THREAD

**Discussion by myself148** [ultralytics/ultralytics#21410] [Q&A] [answered]
<img width="828" height="346" alt="image" src="https://github.com/user-attachments/assets/831a4eda-b060-4e0b-8b95-20240631fc1a" />
官方给出的OBB精度：YOLO11n-obb是78.4，但我从官方下载的yolo11n-OBB进行验证时精度时61.8，下方是我的验证代码，数据集和官方的数据集相同均为DOTA1.0，且是在跑验证集时下载的
import warnings
warnings.filterwarnings('ignore')

from ultralytics import YOLO
import os

# YOLO11n-obb	1024	78.4	117.6 ± 0.8	4.4 ± 0.0	2.7	17.2                     61.8
# YOLO11s-obb	1024	79.5	219.4 ± 4.0	5.1 ± 0.0	9.7	57.5                    65.8
# YOLO11m-obb	1024	80.9	562.8 ± 2.9	10.1 ± 0.4	20.9	183.5               67.4
# YOLO11l-obb	1024	81.0	712.5 ± 5.0	13.5 ± 0.6	26.2	232.0               67.7
# YOLO11x-obb	1024	81.3	1408.6 ± 7.7	28.6 ± 1.0	58.8	520.2           69.3
def main():
    # 加载模型
    model = YOLO("yolo11x-obb.pt")

    # 验证模型
    metrics = model.val(
        data="ultralytics/cfg/datasets/DOTAv1.yaml",
        imgsz=1024,
        batch=2,
        workers=0,
        conf=0.001,
        iou=0.65,
        verbose=True
    )

    print(f"mAP50-95: {metrics.box.map:.4f}")
    print(f"mAP50: {metrics.box.map50:.4f}")
    print(f"mAP75: {metrics.box.map75:.4f}")
    print("Per-class mAP50-95:", metrics.box.maps)

if __name__ == '__main__':
    import multiprocessing
    multiprocessing.freeze_support()  # 防止 Windows 中的多进程报错
    main()
希望有大佬可以帮忙看一下是否有遗漏的参数未写出

**Comment by UltralyticsAssistant** [comment]
👋 Hello @myself148, thank you for sharing your results and providing detailed context! 🚀 This is an automated response to help guide your discussion. An Ultralytics engineer will review your question and assist you soon.

We recommend checking the [Docs](https://docs.ultralytics.com/) for guidance on [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage. Many frequently asked questions and troubleshooting tips can be found there.

If your post is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) (MRE) if you haven’t already. This helps us quickly identify and resolve any issues.

For custom training or validation ❓ questions, please include as much detail as possible, such as dataset samples, config files, and logs. Make sure to follow our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) to optimize your experiments.

Join the Ultralytics community where it suits you best:
- Real-time chat: [Discord](https://discord.com/invite/ultralytics) 🎧
- In-depth topics: [Discourse](https://community.ultralytics.com/)
- Community sharing: [Subreddit](https://www.reddit.com/r/Ultralytics/)

## Upgrade

Ensure you’re using the latest `ultralytics` package and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO runs reliably in these verified environments (with [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are passing. These cover all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu daily and on each commit.

Thank you for contributing and helping to improve Ultralytics!

**Reply by glenn-jocher** [reply, marked as answer]
The difference you're seeing is expected - the official metrics (78.4 mAP for YOLO11n-obb) are **test set results** after submission to the DOTA evaluation server, while your validation code runs on the validation split.

To reproduce the official results, you need to use `split=test` and submit the merged results to the [DOTA evaluation server](https://captain-whu.github.io/DOTA/evaluation.html) as mentioned in the [OBB documentation](https://docs.ultralytics.com/tasks/obb/):

```python
metrics = model.val(
    data="ultralytics/cfg/datasets/DOTAv1.yaml",
    imgsz=1024,
    split="test",  # Use test split instead of val
    device=0
)
```

Your validation split results (61.8) are normal and expected to be lower than the official test results. The official benchmarks use single-model multiscale testing on the test set with results submitted to the DOTA evaluation platform.

**Reply by myself148** [reply]
Hello, thanks for your reply, regarding the single-model multi-scale testing you mentioned, could you please elaborate on the standards for multi-scale testing? Additionally, I found a file named 'test_info.json' on the official DOTA website, and I wonder if you know the function of this file, as well as the roles of parameters such as 'file_name' and 'ins_file_name' in this JSON file.

**Reply by glenn-jocher** [reply]
The multiscale testing mentioned in the [DOTA evaluation documentation](https://docs.ultralytics.com/datasets/obb/dota-v2/) refers to evaluating the model at different image scales. Based on the DOTA preprocessing, this typically involves scales of [0.5, 1.0, 1.5] as seen in the dataset splitting process.

However, for the specific details about `test_info.json` and the parameters like `file_name` and `ins_file_name`, these are part of the official DOTA evaluation server submission format rather than the Ultralytics implementation. You'll need to refer to the [official DOTA evaluation documentation](https://captain-whu.github.io/DOTA/evaluation.html) for the exact specifications of these files and parameters.

The Ultralytics validation process handles the model evaluation, but the final submission format and file structure requirements are defined by the DOTA evaluation server itself.
